### DISCUSSION TITLE
React Hook Form: defaultvalues or setValue?

### DISCUSSION THREAD

**Discussion by gags88** [react-hook-form/react-hook-form#3147] [Q&A]
Hi Fellow Devs,

Just need to clarify that what is optimized for performance?
Say, i am using same form for Edit and Create.

Now, while using Edit function, i may use defaultValue or setValue.

for an eg.

While create operation, i may use:
`const obj = {
  key1: '',
   key2: ''
}`

and while edit i may use
`const obj = {
  key1: value1,
  key2: value2
}`

OR
while edit i can also use
`Object.keys(obj).map(ob => {
    setValue(ob, obj[ob]);
});`

Just want to know what is preferred way and what is optimized for performance.

**Comment by bluebill1049** [comment]
maybe `reset` is what you after.
